src/components/Slider/Slider.js

```diff
@@ -43,7 +43,7 @@ export default class Slider extends PureComponent {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps !== this.props) {
+    if (JSON.stringify(nextProps) !== JSON.stringify(this.props)) {
```

Comment:
Instead of doing a `JSON.stringify` call here, could we use just do direct comparison on the props?

Reply:
I just meant something similar to lodash's `isEqual` method, or fbjs's `areEqual` method. I didn't realize how large the props surface area is for this component though, so my recommendation seems completely inadequate haha.

Reply:
Good discussion here, great that we seriously consider performance! There are some other considerations with walking through objects with `JSON.stringify()` API:

* `JSON.stringify()` throws if the given object contains circular references (Step 1 in: http://www.ecma-international.org/ecma-262/6.0/#sec-serializejsonobject)
* `JSON.stringify()` ignores function refs and symbols (Step 11 and 12 in: http://www.ecma-international.org/ecma-262/6.0/#sec-serializejsonproperty)

One approach we may take is using [`replacer` callback](http://www.ecma-international.org/ecma-262/6.0/#sec-json.stringify), where we can at least relegate walking part to the native API. Walking though and comparing object properties all by JavaScript may be slow especially with non-modern browsers, while IIUC modern browsers has made basic ops (esp. if they are non-layout-affecting and non-memory-instensive) very fast.